packages/turf-length/index.ts

```diff
@@ -33,5 +33,5 @@ export default function length(
       return previousValue! + distance(coords[0], coords[1], options);
     },
     0
-  );
+  ) as number;
```

Comment:
I'm not sure we can avoid the undefined.  Mental exercise for me, for each of these reduce functions (geomReduce, segmentReduce, etc.) there's a *Each function getting called, and if the geojson you pass in has nothing to iterate (empty feature collection for example), and you don't provide an initial value, then the result will be undefined.

Reply:
These casts to number are the only thing I suggest to be looked closer at.  Otherwise this PR looks good to go. I just haven't had time to look.

I think a test with a feature collection with an empty feature array, and a test with features will null geometries could expose the undefined return value, in which case the length can just be returned as zero?

Reply:
@mfedderly Looking further I see that many of the existing *Reduce function calls in `packages/turf-meta/types.ts` among other things are currently in error (red squiggles in vs code) because they expect the return value to be `number` and don't acknowledge undefined.  The code in types.ts files aren't official tests so they don't surface in CI but I think they should still get cleaned up, at least acknowledging the rough edges.  I'm happy to approve once that's done.  I didn't have enough time today to take a shot at it.

![image](https://user-images.githubusercontent.com/81816/200134296-b0efe258-4cc9-4820-b896-b5f5eb318681.png)

I think that's a way forward for now, but I don't think that's the full story.  I think the cause of the `T | undefined` getting returned for all these *Reduce functions is the fact that the default value is optional.  I think that function overloading is a potential way to handle this, such that if you provide a default value that is a `number` it knows that the return type is `number`.  Related - https://stackoverflow.com/questions/70621041/empty-optional-parameter-not-inferred-as-undefined.  This could be explored in a follow-up PR.  I think better test coverage for not passing a default value to these functions is needed too.